src/threadpool.c

```diff
@@ -261,6 +268,62 @@ static void init_once(void) {
   init_threads();
 }
 
+int uv_set_threadpool_size(unsigned int number) {
+  unsigned int spin_nthreads;
+
+  /* Ensure the threadpool is initialized */
+  if (nthreads == 0)
+    uv_once(&once, init_once);
+
+  /* no-op if the threadpool size is already set */
+  if (number == nthreads)
```

Comment:
What if 2 threads are calling this function at the same time? You might need a mutex.

Reply:
I wasn’t considering this to be thread safe 🫠🥲 but yeah, I can work on that

Reply:
Yeah you do need the mutex since you are mutating the global state.